junebug/api.py

```diff
@@ -299,5 +301,44 @@ def get_message_status(self, request, channel_id, message_id):
         }))
 
     @app.route('/health', methods=['GET'])
+    @inlineCallbacks
     def health_status(self, request):
-        return response(request, 'health ok', {})
+        if self.config.rabbitmq_management_interface:
+            rabbit_client = Client(self.config.rabbitmq_management_interface,
+                                   self.amqp_config['username'],
+                                   self.amqp_config['password'])
+
+            ids = yield Channel.get_all(self.redis)
+
+            queues = []
+            stuck = False
+
+            for channel_id in ids:
+                for sub in ['inbound', 'outbound', 'event']:
+                    queue_name = "%s.%s" % (channel_id, sub)
+
+                    queue = rabbit_client.get_queue(
+                        self.amqp_config['vhost'], queue_name)
```

Comment:
This is doing a blocking request, which is very bad in twisted land. If there's no way to have this client do a non-blocking request, or if there aren't any non blocking clients, then we might have to create our own (we probably only need to implement this one endpoint for now though)